### DISCUSSION TITLE
The configuration options placed in devcontainer.json not getting picked

### DISCUSSION THREAD

**Discussion by ayush987goyal** [community/community#26870] [Codespaces]
<p>Here is the config I used:</p>
<pre><code class="lang-auto">{
    "name": "Dev Container Definition - AWS CDK",
    "image": "jsii/superchain",
    "postCreateCommand": "yarn build --skip-test --no-bail --skip-prereqs --skip-compat",
    "extensions": [
        "dbaeumer.vscode-eslint@2.1.5:9Wg0Glx/TwD8ElFBg+FKcQ=="
    ]
}
</code></pre>
<p>Here is the codespace for the same: <a href="https://ayush987goyal-aws-cdk-g78p.github.dev/" rel="noopener nofollow ugc">https://ayush987goyal-aws-cdk-g78p.github.dev/</a></p>
<p>The codespace was not created by the provided image and the <code>postCreateCommand</code> didn’t run as well.</p>

**Comment by lostintangent** [comment]
<p>Hey! Could you confirm where you’re storing the devcontainer.json file at in your repo? Also, you mentioned that the image and postCreateCommand didn’t work. Did the ESLint extension get installed as expected though?</p>

**Comment by ayush987goyal** [comment]
<p>Hey! The devcontainer.json file is stored in the root of the project. Now that you mentioned it, I noticed that the extension didn’t get installed either. Here is the repo for your reference:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/ayush987goyal/aws-cdk/tree/pr/dev-container" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/5697227/181100392-2e848ab8-d40e-4e19-a286-7a537e85ba07.jpeg" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/ayush987goyal/aws-cdk/tree/pr/dev-container" target="_blank" rel="noopener nofollow ugc">ayush987goyal/aws-cdk</a></h3>

<p>The AWS Cloud Development Kit is a framework for defining cloud infrastructure in code - ayush987goyal/aws-cdk</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by jkeech** [comment, marked as answer]
<p>It looks like there’s a small typo in the name of the file. If it’s at the root of the repo, it needs to be called <code>.devcontainer.json</code> with a leading period instead of <code>devcontainer.json</code>. If it’s inside of a <code>.devcontainer</code> folder, then it should be called <code>devcontainer.json</code> without a leading period. I do think this naming convention is confusing though, so you’re probably not the first to run into this issue.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://code.visualstudio.com/favicon.ico" class="site-icon" width="256" height="256">
      <a href="https://code.visualstudio.com/docs/remote/containers#_create-a-devcontainerjson-file" target="_blank" rel="noopener nofollow ugc">code.visualstudio.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/746020/181100394-86077102-b97c-4d89-8a29-ae1019a98c7e.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/fb44b7af244eca55a27ec6831e612d88eebbaafa_2_690x345.png, https://aws1.discourse-cdn.com/github/original/2X/f/fb44b7af244eca55a27ec6831e612d88eebbaafa.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/fb44b7af244eca55a27ec6831e612d88eebbaafa.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/fb44b7af244eca55a27ec6831e612d88eebbaafa_2_10x10.png"></div>

<h3><a href="https://code.visualstudio.com/docs/remote/containers#_create-a-devcontainerjson-file" target="_blank" rel="noopener nofollow ugc">Developing inside a Container using Visual Studio Code Remote Development</a></h3>

<p>Developing inside a Container using Visual Studio Code Remote Development</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>It looks like there’s actually a documentation bug at <a href="https://docs.github.com/en/github/developing-online-with-codespaces/configuring-codespaces-for-your-project#about-default-codespace-configurations" rel="noopener nofollow ugc">https://docs.github.com/en/github/developing-online-with-codespaces/configuring-codespaces-for-your-project#about-default-codespace-configurations</a>, which doesn’t show the leading period (<a class="mention" href="/matthewisabel">@matthewisabel</a> FYI).</p>

**Comment by ayush987goyal** [comment]
<p>Thanks.</p>
<ol>
<li>Does the <code>postCreateCommand</code> run in background if the codespace is closed/timed-out? (I have seen multiple occurrences where the command didn’t finish and they codespace was disconnected and now there is no way to run the command again or check the logs )</li>
</ol>

**Comment by jkeech** [comment]
<p><a class="mention" href="/ayush987goyal">@ayush987goyal</a>, yes the command runs async in the background. It’s not tied to whether you are connected or not. If it’s still running when you are connected, you should see a toast notification letting you know. If you don’t see the toast, or if you reconnect in the future and want to see the logs, you can run the command <code>Codespaces: View Creation Log</code> or view the Codespace in the Remote Explorer viewlet and right click and select View Creation Log.</p>

**Comment by ayush987goyal** [comment]
<p>Thanks a lot <a class="mention" href="/jkeech">@jkeech</a></p>

**Comment by ayush987goyal** [comment]
<p><span class="mention">@jkeechI</span> have also noticed that:</p>
<ol>
<li>The creation logs does not update in realtime when view by running the <code>Codespaces: View Creation Log</code>
</li>
<li>The extensions get auto-updated even though we mention a specific version in <code>.devcontainer.json</code>
</li>
<li>Every time a workspace is created, the extensions are installed based on the user-preferences. Some of these extensions require a reload before which they can be activated. For example, extension like color and icon theme are set in settings but they don’t take effect until after we restart the codespace for extensions to take effect. Could we automatically reload if some preferred extension require them?</li>
</ol>

**Comment by jkeech** [comment]
<p><a class="mention" href="/ayush987goyal">@ayush987goyal</a>,</p>
<p>For (1), the logs are supposed to be updating live, unless the scripts are finished. In the case where they are finished, it just shows the output of the log files from when the commands ran. Are you seeing a delay or just not updating at all? Is it possible your script finished or is somehow hanging indefinitely with no output?</p>
<p>For (2), I don’t think pinning the version is supported in devcontainer.json. It always downloads the latest version from the marketplace based on the extension id.</p>
<p>For (3), VS Code is working on getting hot-reload of extensions to work in the web like is already does on the desktop. So in the next month or so that will hopefully work automatically without needing to reload. You can always reload the browser manually today.</p>
<p>It might be easier to start new posts if you run into additional issues so that each post is scoped to a specific problem when other people search for questions/answers :slight_smile:</p>
